lib/collections/users.js

```diff
@@ -133,9 +133,66 @@ function checkParentGuardian(age, parentGuardian) {
   }
 }
 
+// Check if the country entered is for the United States.
+function isUnitedStates(country) {
+    let c = country.toUpperCase();
+    return ((c == 'UNITED STATES') ||
+	    (c == 'USA') ||
+	    (c == 'U.S.') ||
+	    (c == 'U.S.A.') ||
+	    (c == 'US'));
+}
+
+// Check if the country entered is for Canada
+function isCanada(country) {
+    let c = country.toUpperCase();
+    return ((c == 'CANADA') || (c == 'CA') || (c == 'CAN'));
+}
+
+// State values are required for the US and Canada; not checked
+// for other countries
+function checkCountryState(data) {
+    let country = data.country;
+    if (isUnitedStates(country) || isCanada(country)) {
+	if ((data.state == '') || (data.state == 'OTHER')) {
+	    throw new Meteor.Error(400, 'State/province required for US and Canadian addresses');
+	}
+    }
+}
+
+// If this is a US address, check that the zip code conforms to a zip
+// or zip+4 pattern. If this is a Canadian address, check that the
+// postcode conforms to the Canadian code pattern. Do not check other
+// countries.
+function checkPostalCode(country, postcode) {
+    let okay = true;
+    if (isUnitedStates(country)) {
+	if (postcode.length == 0) {
```

Comment:
Are we guaranteed that `postcode` will always be a string here? Technically the verification on line 86 calls it an "any" type so it may not have a length property. Checking for `!postcode`, or else ensuring it's a string within this function, would probably make this function safer if something changes down the road.

Reply:
Made suggested change.